### PR TITLE
Do not infinitly check for depexts when the system package manager fails

### PR DESCRIPTION
**Pull request by kit-ty-kate**
```
[1/2/3/4] 1
+ /usr/bin/sudo "apt-get" "install" "-qq" "-yy" "autoconf" "gnuplot-x11"
[ERROR] System package install failed with exit code 100 at command:
            sudo apt-get install -qq -yy autoconf gnuplot-x11
- E: Unable to correct problems, you have held broken packages.
[ERROR] These packages are still missing: autoconf gnuplot-x11

opam believes some required external dependencies are missing. opam can:
> 1. Run apt-get to install them (may need root/sudo access)
  2. Display the recommended apt-get command and wait while you run it manually 
(e.g. in another terminal)
  3. Attempt installation anyway, and permanently register that this external de
pendency is present, but not detectable
  4. Abort the installation

[1/2/3/4] 1
+ /usr/bin/sudo "apt-get" "install" "-qq" "-yy" "autoconf" "gnuplot-x11"
- E: Unable to correct problems, you have held broken packages.
[ERROR] System package install failed with exit code 100 at command:
            sudo apt-get install -qq -yy autoconf gnuplot-x11
[ERROR] These packages are still missing: autoconf gnuplot-x11

opam believes some required external dependencies are missing. opam can:
> 1. Run apt-get to install them (may need root/sudo access)
  2. Display the recommended apt-get command and wait while you run it manually 
(e.g. in another terminal)
  3. Attempt installation anyway, and permanently register that this external de
pendency is present, but not detectable
  4. Abort the installation

[1/2/3/4] 1
+ /usr/bin/sudo "apt-get" "install" "-qq" "-yy" "autoconf" "gnuplot-x11"
- E: Unable to correct problems, you have held broken packages.
[...]
```

### CODE REVIEW

**Review by rjbou — APPROVED** [review]

**Comment by kit-ty-kate** [comment]
Waiting on https://github.com/ocaml/opam/pull/5268 to add a testcase to avoid future regression

File: tests/reftests/depexts.test [hunk 1]
```diff
@@ -0,0 +1,41 @@
+N0REP0
+### <pkg:a.1>
+opam-version: "2.0"
+depexts: "depext-1"
+### OPAMVAR_os_family=dummy-failure OPAMNODEPEXTS=0
```
Comment (rjbou) [inline tests/reftests/depexts.test:5]: why not opam option?

File: tests/reftests/depexts.test [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  4. Abort the installation
+
+[1/2/3/4] 1
+
+false 
+[ERROR] System package install failed with exit code 1 at command:
+            false
+This command should get the requirements installed:
+
+    false
```
Comment (rjbou) [inline tests/reftests/depexts.test:30]: `false` should have the package as argument, but it is not the topic of this PR.

File: tests/reftests/run.ml [hunk 3]
```diff
@@ -418,16 +418,17 @@ module Parse = struct
...
             let re =
               seq [
+                Re.str "+ ";
                 rep any;
                 alt [ set "/\\\"" ];
                 Re.str cmd;
                 opt @@ Re.str ".exe";
                 opt @@ char '"';
-                Re.str " \""
+                Re.str " "
```
Comment (rjbou) [inline tests/reftests/run.ml:427]:
```suggestion
                Re.space
```

File: tests/reftests/depexts.test [hunk 4]
```diff
@@ -0,0 +1,41 @@
+N0REP0
+### <pkg:a.1>
+opam-version: "2.0"
+depexts: "depext-1"
+### OPAMVAR_os_family=dummy-failure OPAMNODEPEXTS=0
+### opam switch create test --empty
+### opam install a | sed-cmd false
```
Comment (rjbou) [inline tests/reftests/depexts.test:7]:
```suggestion
### : Erroneous package manager install is not blocking :
### opam install a | sed-cmd false
```

File: tests/reftests/run.ml [hunk 5]
```diff
@@ -418,16 +418,17 @@ module Parse = struct
...
               ]
-(*                 Printf.sprintf ".*(/|\\\\|\")%s(\\.exe)?\"? \"" cmd *)
+(*                 Printf.sprintf ".*(/|\\\\|\")%s(\\.exe)?\"? " cmd *)
```
Comment (rjbou) [inline tests/reftests/run.ml:429]:
```suggestion
(*                 Printf.sprintf "+ .*(/|\\\\|\")%s(\\.exe)?\"? " cmd *)
```

File: tests/reftests/depexts.test (hunk 1 above)
Comment (rjbou) [reply tests/reftests/depexts.test:5]:
```suggestion
### OPAMNODEPEXTS=0
### opam var --global os-family=dummy-failure
Added '[os-family "dummy-failure" "Set through 'opam var'"]' to field global-variables in global configuration
```

**Review by rjbou — APPROVED** [review]
Some tiny things about the test, otherwise lgtm

File: tests/reftests/run.ml (hunk 3 above)
Comment (kit-ty-kate) [reply tests/reftests/run.ml:427]: Re.space does not have the same behaviour as `str " "` (the first one accepts other space characters that we should not accept)

**Pull request merged**